### PR TITLE
HCAL: bug fixed in DPGAnalysis/HcalTools for 2023 Run3

### PR DESCRIPTION
**Pull request by zhokin2**
#### PR description:

this is an additional PR to overcome serious warning seen only with command " scram b checker " and concerns only file plugin/[CMTRawAnalyzer.cc] (https://github.com/cms-sw/cmssw/blob/master/DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc)

#### PR validation:

scram build code-format

scram b checker    

scram build code-checks

### CODE REVIEW

File: DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc [hunk 1]
```diff
@@ -8821,23 +8818,23 @@ void CMTRawAnalyzer::fillDigiAmplitude(HBHEDigiCollection::const_iterator& digiI
   pedestalaver4 /= c4;
   pedestalwaver9 = sqrt(pedestalwaver9 / TSsize);
   pedestalwaver4 = sqrt(pedestalwaver4 / c4);
-  if (ts_with_max_signal > -1 && ts_with_max_signal < 10)
+  if (ts_with_max_signal > 0 && ts_with_max_signal < TSsize)
```
Comment (perrotta) [inline DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc:8821]: Why here and in the following you have "> 0" and you have "> -1" in all other cases?

File: DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc [hunk 2]
```diff
@@ -9662,7 +9659,7 @@ void CMTRawAnalyzer::fillDigiAmplitudeQIE11(QIE11DataFrame qie11df) {
   double amplitude345 = 0.;
   double ampl = 0.;
   double ampl3ts = 0.;
-  double amplmaxts = 0.;
+  //  double amplmaxts = 0.;
```
Comment (perrotta) [inline DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc:9662]: Please remove the commented out code with these useless variables, here and below

File: DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc (hunk 1 above)
Comment (zhokin2) [reply DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc:8821]: in principle, it does not matter, but -1 is more correct, Thank you !

File: DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc (hunk 2 above)
Comment (zhokin2) [reply DPGAnalysis/HcalTools/plugins/CMTRawAnalyzer.cc:9662]: OK, these commented lines can be removed - thanks.

**Comment by zhokin2** [comment]
tiny corrections of plugins/CMTRawAnalyzer.cc done with following
git commit -m "HCAL: bug fixed in DPGAnalysis/HcalTools for 2023 Run3" ./
git push my-cmssw CMTRMTforRun3_CMSSW_13_0_X_2023-02-08-1100

________________________________

**Comment by emanueleusai** [comment]
type hcal

**Comment by emanueleusai** [comment]
type bug

**Comment by emanueleusai** [comment]
please test

**Comment by perrotta** [comment]
@zhokin2 there are still commented out lines to remove:
- [L9661](https://github.com/cms-sw/cmssw/pull/40732/files#diff-b578bd20219108c2b5aa1fcd0f0f6a011ec4c285141493d777a2bfc752d254e1R9661)
- [L9788](https://github.com/cms-sw/cmssw/pull/40732/files#diff-b578bd20219108c2b5aa1fcd0f0f6a011ec4c285141493d777a2bfc752d254e1R9788)
- [L9808](https://github.com/cms-sw/cmssw/pull/40732/files#diff-b578bd20219108c2b5aa1fcd0f0f6a011ec4c285141493d777a2bfc752d254e1R9808)

**Comment by zhokin2** [comment]
3 comments removed (L9661, L9788, L9808)

________________________________

**Comment by perrotta** [comment]
Please test

**Comment by perrotta** [comment]
@zhokin2 @abdoulline this PR is not in conflict with https://github.com/cms-sw/cmssw/pull/40721 (which I think was the conclusion you reached in the thread ending with https://github.com/cms-sw/cmssw/pull/40721#issuecomment-1425928143)
If you closed this PR because of some different reason, instead, feel free to close it again

**Comment by perrotta** [comment]
> frankly, there is no idea what is the problem in runTheMatrix, because in the last edit, I just deleted three commented lines

There are no problems at all in runTheMatrix, as far as I can see. RelVals-INPUT time out quite often, independently on the PR. When it is clear, as in this case, that the timeour cannot be given by the PR changes we don't bother to re-run them. But if you like seeing the display all green, we can relaunch the tests here and see whether those timeouts disappear by themselves.

**Comment by perrotta** [comment]
please test

**Comment by abdoulline** [comment]
@perrotta   I'm a bit confused... No conflict, but PRs modify the same file CMTRawAnalyzer.cc ?

**Comment by perrotta** [comment]
> @perrotta I'm a bit confused... No conflict, but PRs modify the same file CMTRawAnalyzer.cc ?

git can recognize that the modifications appeared in different parts of the file, far away "enough" according to some metrics, and considers them as independent.
In fact, #40721 removed some unused variable, while this PR touches other variables in other parts of the code. As such the modifications applied are independent on each other, and can be integrated independently. 
If you feel more confortable, you can also rebase this PR on top of the latest IB: but looking at the code changes, I don't think it is necessary.

**Comment by abdoulline** [comment]
@perrotta   OK I see now.

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
+1
- The file touched by this PR is used for HCAL local monitoring, and doesn't touch any other workflow: therefore it can be merged in 13_0_X even before switching the master to 13_1_X, to make it already available also in 13_0_X for Run3 monitoring

**Pull request merged**